src/util/newconfig/ConfigFileJson.cpp

```diff
@@ -0,0 +1,157 @@
+    purpose with or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL,  DIRECT,  INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include "util/newconfig/ConfigFileJson.hpp"
+
+#include "util/Assert.hpp"
+#include "util/log/Logger.hpp"
+#include "util/newconfig/Errors.hpp"
+#include "util/newconfig/Types.hpp"
+
+#include <boost/filesystem/path.hpp>
+#include <boost/json/array.hpp>
+#include <boost/json/object.hpp>
+#include <boost/json/parse.hpp>
+#include <boost/json/parse_options.hpp>
+#include <boost/json/value.hpp>
+#include <fmt/core.h>
+
+#include <cstddef>
+#include <exception>
+#include <fstream>
+#include <ios>
+#include <iostream>
+#include <optional>
+#include <ostream>
+#include <sstream>
+#include <string>
+#include <string_view>
+#include <vector>
+
+namespace util::config {
+
+void
+ConfigFileJson::parse(boost::filesystem::path configFilePath)
+{
+    try {
+        std::ifstream const in(configFilePath.string(), std::ios::in | std::ios::binary);
+        if (in) {
+            std::stringstream contents;
+            contents << in.rdbuf();
+            auto opts = boost::json::parse_options{};
+            opts.allow_comments = true;
+            auto const tempObj = boost::json::parse(contents.str(), {}, opts).as_object();
+            flattenJson(tempObj, "");
+        }
+    } catch (std::exception const& e) {
+        LOG(util::LogService::error()) << "Could not read configuration file from '" << configFilePath.string()
+                                       << "': " << e.what();
+    }
+}
+
+/**
+ * @brief Extracts the value from a JSON object and converts it into the corresponding type.
+ *
+ * @param jsonValue The JSON value to extract.
+ * @return A variant containing the same type corresponding to the extracted value.
+ */
+[[nodiscard]] static Value
+extractJsonValue(boost::json::value const& jsonValue)
+{
+    Value variantValue;
+
+    if (jsonValue.is_int64()) {
+        variantValue = jsonValue.as_int64();
+    } else if (jsonValue.is_string()) {
+        variantValue = jsonValue.as_string().c_str();
+    } else if (jsonValue.is_bool()) {
+        variantValue = jsonValue.as_bool();
+    } else if (jsonValue.is_double()) {
+        variantValue = jsonValue.as_double();
+    }
+    return variantValue;
+}
+
+Value
```

Comment:
There's actually no point on checking whether key exists because there is a [separate check before the call to these get functions](https://github.com/XRPLF/clio/pull/1593/files#diff-4e8d0eeb9ae15e14a017e6d321f3987f1c1e411fd58bb6eea9540f715328818eR236). I removed the asserts.